components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java

```diff
@@ -0,0 +1,263 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.BRANDING_RESOURCE_TYPE;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_ERROR_BUILDING_BRANDING_PREFERENCE;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_ERROR_GETTING_BRANDING_PREFERENCE;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.ORGANIZATION_TYPE;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.RESOURCE_NAME_SEPARATOR;
+import static org.wso2.carbon.identity.branding.preference.management.core.constant.BrandingPreferenceMgtConstants.RESOURCE_NOT_EXISTS_ERROR_CODE;
+import static org.wso2.carbon.identity.branding.preference.management.core.util.BrandingPreferenceMgtUtils.handleClientException;
+import static org.wso2.carbon.identity.branding.preference.management.core.util.BrandingPreferenceMgtUtils.handleServerException;
+
+/**
+ * UI Branding Preference Resolver Implementation.
+ */
+public class UIBrandingPreferenceResolverImpl implements UIBrandingPreferenceResolver {
+
+    private static final Log LOG = LogFactory.getLog(UIBrandingPreferenceResolverImpl.class);
+
+    private final BrandedOrgCache brandedOrgCache;
+
+    /**
+     * UI branding preference resolver implementation constructor.
+     *
+     * @param brandedOrgCache Cache instance
+     */
+    public UIBrandingPreferenceResolverImpl(BrandedOrgCache brandedOrgCache) {
+
+        this.brandedOrgCache = brandedOrgCache;
+    }
+
+    @Override
+    public BrandingPreference resolveBranding(String type, String name, String locale)
+            throws BrandingPreferenceMgtException {
+
+        String organizationId = getOrganizationId();
+        String currentTenantDomain = getTenantDomain();
+
+        OrganizationManager organizationManager =
+                BrandingResolverComponentDataHolder.getInstance().getOrganizationManager();
+
+        /* Tenant domain will always be carbon.super for SaaS apps (ex. myaccount). Hence need to resolve
+          tenant domain from the name parameter. */
+        if (ORGANIZATION_TYPE.equals(type) &&
+                MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(currentTenantDomain)) {
+            currentTenantDomain = name;
+            try {
+                organizationId = organizationManager.resolveOrganizationId(currentTenantDomain);
```

Comment:
As the current tenant domain is equals to carbon.super, it is not required to resolve the orgId. you can use
SUPER_ORG_ID constant.

Reply:
Not related - https://github.com/wso2-extensions/identity-branding-preference-management/pull/15/files#diff-8fb3ee38c3d4a60f10e215ddc7c2787da799305367f9487df546ccfe4d7ae714R93